cabra/src/pages/components/RecurrenceForm.tsx

```diff
@@ -0,0 +1,150 @@
+import "twin.macro";
+
+import { PossibleEnds, PossibleFrequency } from "../../types/recurrenceRule";
+import parseRecurrenceRule, {
+  DEFAULT_RULE,
+  serializeRecurrenceRule,
+} from "../../utils/rrule";
+import tw, { styled } from "twin.macro";
+import { useEffect, useState } from "react";
+
+import { DATE_FORMAT } from "../../utils/dates";
+import DatePicker from "react-datepicker";
+
+type Props = {
+  onChange: (rrule: string | null) => void;
+  value: string | null;
+};
+
+const endsKeys = ["NEVER", "COUNT", "UNTIL"] as const;
+const endText = {
+  [PossibleEnds.NEVER]: "Never",
+  [PossibleEnds.COUNT]: "After",
+  [PossibleEnds.UNTIL]: "On date",
+};
+const frequencyKeys = ["DAILY", "WEEKLY", "MONTHLY"] as const;
+const frequencyText = {
+  [PossibleFrequency.DAILY]: "day(s)",
+  [PossibleFrequency.WEEKLY]: "week(s)",
+  [PossibleFrequency.MONTHLY]: "month(s)",
+};
+
+const inputStyles = tw`rounded-lg text-black bg-tertiary px-4 py-2`;
+
+const Input = styled.input(inputStyles);
+const Select = styled.select(inputStyles, tw`capitalize`);
+
+function RecurrenceForm({ onChange, value }: Props) {
+  const [rule, setRule] = useState(() =>
+    value !== null ? parseRecurrenceRule(value) : DEFAULT_RULE
+  );
+  const [isRecurrent, setIsRecurrent] = useState(value !== null);
+
+  useEffect(() => {
+    if (!isRecurrent) onChange(null);
+    else onChange(serializeRecurrenceRule(rule));
+  }, [isRecurrent, onChange, rule]);
+
+  return (
+    <div className="text-stone-50">
+      <div tw="flex items-center">
+        <label
+          htmlFor="recurrence-checkbox"
+          tw="text-gray-50 font-bold space-x-2 "
+        >
+          Recurrent Task:
+        </label>
+        <input
+          checked={isRecurrent}
+          id="recurrence-checkbox"
+          name="recurrence-checkbox"
+          onChange={() => setIsRecurrent((r) => !r)}
+          type="checkbox"
+          tw="ml-2"
+        />
+      </div>
+      {isRecurrent && (
+        <div tw="text-white font-semibold mt-2 space-y-2">
+          <div tw="flex items-center gap-4">
+            <label htmlFor="repeat">Repeat</label>
+            <Select
+              name="repeat"
+              tw="w-28"
+              value={rule.frequency}
+              onChange={(e) =>
+                setRule((r) => ({ ...r, frequency: +e.target.value }))
```

Comment:
Using "+" to cast to a number is not recommended in Airbnb style guide. See: https://github.com/airbnb/javascript#coercion--numbers

But it can stay as it is